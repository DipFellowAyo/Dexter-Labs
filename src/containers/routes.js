import React from 'react'
import { Redirect, Route, Switch } from 'react-router-dom'

import {
  ADMIN_AUTH,
  LANDING_ROUTE,
  FORGOT_PASSWORD_ROUTE,
  LOGIN_ROUTE,
  MERCHANTS_ROUTE,
  NOT_FOUND_ROUTE,
  PASSWORD_PERSONALIZATION_ROUTE,
  PASSWORD_RESET_SUCCESSFULL_ROUTE,
  REGISTER_ROUTE,
  REGISTRATION_SUCCESSFUL_ROUTE,
  TRANSACTIONS_ROUTE,
  DAASHBOARD_ROUTE,
  MESSAGES_ROUTE,
  DSA_DASHBOARD_ROUTE
} from '../utils/routes'

const Landing = React.lazy(() => import('../pages/landing'))
const Login = React.lazy(() => import('../pages/login'))
const Register = React.lazy(() => import('../pages/register'))
const ForgotPassword = React.lazy(() => import('../pages/forgot-password'))
const PasswordResetForgotSuccessful = React.lazy(() =>
  import('../pages/password-reset-forgot-successful')
)
const RegistrationSuccessful = React.lazy(() =>
  import('../pages/registration-successful')
)
const AdminAuth = React.lazy(() => import('../pages/admin-auth'))
const PasswordPersonalization = React.lazy(() =>
  import('../pages/password-personalization')
)
const Dashboard = React.lazy(() => import('../pages/dashboard/index'))
const Merchants = React.lazy(() => import('../pages/dashboard/merchants'))
const Transactions = React.lazy(() => import('../pages/dashboard/transactions'))
const Messages = React.lazy(() => import('../pages/dashboard/messages'))
const NotFound = React.lazy(() => import('../pages/not-found'))
const DSA_Dashboard = React.lazy(() => import('../pages/DSA_dashboard/index'))

const Routes = () => {
  return (
    <Switch>
      <Route exact path={LANDING_ROUTE} component={Landing} />
      <Route exact path={LOGIN_ROUTE} component={Login} />
      <Route exact path={REGISTER_ROUTE} component={Register} />
      <Route exact path={FORGOT_PASSWORD_ROUTE} component={ForgotPassword} />
      <Route
        exact
        path={PASSWORD_RESET_SUCCESSFULL_ROUTE}
        component={PasswordResetForgotSuccessful}
      />
      <Route
        exact
        path={REGISTRATION_SUCCESSFUL_ROUTE}
        component={RegistrationSuccessful}
      />
      <Route exact path={ADMIN_AUTH} component={AdminAuth} />
      <Route
        exact
        path={PASSWORD_PERSONALIZATION_ROUTE}
        component={PasswordPersonalization}
      />
      <Route exact path={DSA_DASHBOARD_ROUTE} component={DSA_Dashboard} />
      <Route exact path={DAASHBOARD_ROUTE} component={Dashboard} />
      <Route exact path={MERCHANTS_ROUTE} component={Merchants} />
      <Route exact path={TRANSACTIONS_ROUTE} component={Transactions} />
      <Route exact path={MESSAGES_ROUTE} component={Messages} />
      <Route exact path={NOT_FOUND_ROUTE} component={NotFound} />
      <Redirect from='*' to={NOT_FOUND_ROUTE} />
    </Switch>
  )
}

export default Routes
