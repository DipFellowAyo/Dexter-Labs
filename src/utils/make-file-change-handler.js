export default function makeFileChangeHandler ({
  key,
  // setPreview,
  setFieldValue,
  setFieldError,
  setFieldTouched,
  maxFileSize = 1,
  mimeErrorMessage = 'Images only please.',
  fileTypeRegex = /^data:image\/(\w+);base64,/,
  fileSizeErrorMessage = `File size cannot be more than ${maxFileSize}MB.`
}) {
  return e => {
    e.persist()

    if (!e.target.files.length) {
      return setFieldValue(key, '')
    }

    setFieldTouched(key, true, false)

    const file = e.target.files[0]

    const fileSize = file.size / 1024 / 1024

    if (fileSize > maxFileSize) {
      e.target.value = ''

      return setFieldError(key, fileSizeErrorMessage)
    }

    const reader = new window.FileReader()

    reader.readAsDataURL(file)

    reader.onload = () => {
      const dataURI = reader.result

      if (!fileTypeRegex.test(dataURI)) {
        return setFieldError(key, mimeErrorMessage)
      }

      // setPreview(dataURI)

      setFieldValue(key, dataURI)
    }

    reader.onerror = () => setFieldValue(key, '')
  }
}
