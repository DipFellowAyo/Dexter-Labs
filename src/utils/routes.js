export const LANDING_ROUTE = '/'

export const LOGIN_ROUTE = '/login'

export const FORGOT_PASSWORD_ROUTE = '/forgot-password'

export const REGISTER_ROUTE = '/register'

export const PASSWORD_RESET_SUCCESSFULL_ROUTE = '/password-reset-successful'

export const REGISTRATION_SUCCESSFUL_ROUTE = '/registration-successful'

export const ADMIN_AUTH = '/admin-auth'

export const PASSWORD_PERSONALIZATION_ROUTE = '/password-personalization'

export const DSA_DASHBOARD_ROUTE = '/dsa-dashboard'

export const DAASHBOARD_ROUTE = '/dashboard'

export const MERCHANTS_ROUTE = '/dashboard/merchants'

export const TRANSACTIONS_ROUTE = '/dashboard/transactions'

export const MESSAGES_ROUTE = '/dashboard/messages'

export const NOT_FOUND_ROUTE = '/not-found'
