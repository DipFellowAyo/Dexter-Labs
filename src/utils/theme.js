import { createTheme } from '@material-ui/core/styles'

const primaryColor = '#6F6F6F'
const secondaryColor = '#ec7e81'

export const theme = createTheme({
  typography: {
    fontFamily: 'Poppins'
  },
  overrides: {
    MuiCssBaseline: {
      '@global': {
        '@font-face': ['Poppins']
      }
    }
  },
  palette: {
    primary: {
      main: primaryColor
    },
    secondary: {
      main: secondaryColor
    }
  }
})
