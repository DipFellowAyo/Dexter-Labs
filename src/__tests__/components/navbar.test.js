import React from 'react'
import Box from '@material-ui/core/Box'
import IconButton from '@material-ui/core/IconButton'
import toJson from 'enzyme-to-json'
import { Link } from 'react-router-dom'
import { shallow } from 'enzyme'

import NavBar from '../../components/navbar'
import NotificationBadge from '../../components/notification-badge'
import MessageBadge from '../../components/message-badge'
import { LANDING_ROUTE } from '../../utils/routes'

describe('Navbar component', () => {
  let navBar

  beforeEach(() => {
    navBar = shallow(<NavBar />)
  })

  it('renders correctly', () => {
    expect(toJson(navBar)).toMatchSnapshot()
  })

  it('renders a Box component', () => {
    expect(navBar.find(Box).exists()).toBe(true)
  })

  it('renders a Box component with a prop name `component` at node 0', () => {
    expect(
      navBar
        .find(Box)
        .at(0)
        .props().component
    ).toEqual('nav')
  })

  it('renders a Box component with a prop name `className` at node 0', () => {
    expect(
      navBar
        .find(Box)
        .at(0)
        .props().className
    ).toEqual('makeStyles-appBarStyles-1')
  })

  it('renders a Link component', () => {
    expect(navBar.find(Link).exists()).toBe(true)
  })

  it('renders a Link component with a prop name `to`', () => {
    expect(navBar.find(Link).props().to).toEqual(LANDING_ROUTE)
  })

  it('renders a Box component with a prop name `component` at node 1', () => {
    expect(
      navBar
        .find(Box)
        .at(1)
        .props().component
    ).toEqual('img')
  })

  it('renders a Box component with a prop name `src` at node 1', () => {
    expect(
      navBar
        .find(Box)
        .at(1)
        .props().src
    ).toEqual('/vectors/dexter.svg')
  })

  it('renders a Box component with a prop name `alt` at node 1', () => {
    expect(
      navBar
        .find(Box)
        .at(1)
        .props().alt
    ).toEqual('dexter')
  })

  it('renders a Box component with a prop name `className` at node 1', () => {
    expect(
      navBar
        .find(Box)
        .at(1)
        .props().className
    ).toEqual('makeStyles-logo-5')
  })

  it('renders a Box component with a prop name `component` at node 2', () => {
    expect(
      navBar
        .find(Box)
        .at(2)
        .props().component
    ).toEqual('img')
  })

  it('renders a Box component with a prop name `src` at node 2', () => {
    expect(
      navBar
        .find(Box)
        .at(2)
        .props().src
    ).toEqual('/vectors/logo.svg')
  })

  it('renders a Box component with a prop name `alt` at node 2', () => {
    expect(
      navBar
        .find(Box)
        .at(2)
        .props().alt
    ).toEqual('logo')
  })

  it('renders a Box component with a prop name `className` at node 2', () => {
    expect(
      navBar
        .find(Box)
        .at(2)
        .props().className
    ).toEqual('makeStyles-mobile-6')
  })

  it('renders a Box component with a prop name `display` at node 3', () => {
    expect(
      navBar
        .find(Box)
        .at(3)
        .props().display
    ).toEqual('flex')
  })

  it('renders a Box component with a prop name `alignItems` at node 3', () => {
    expect(
      navBar
        .find(Box)
        .at(3)
        .props().alignItems
    ).toEqual('center')
  })

  it('renders a form', () => {
    expect(navBar.find('form').exists()).toBe(true)
  })

  it('render a Box component with a prop name `className` at node 4', () => {
    expect(
      navBar
        .find(Box)
        .at(4)
        .props().className
    ).toBe('makeStyles-searchField-2')
  })

  it('renders an input field', () => {
    expect(navBar.find('input').exists()).toBe(true)
  })

  it('renders an input field with a prop name `type`', () => {
    expect(navBar.find('input').props().type).toEqual('text')
  })

  it('renders an input field with a prop name `placeholder`', () => {
    expect(navBar.find('input').props().placeholder).toEqual('Search')
  })

  it('renders an input field with a prop name `className`', () => {
    expect(navBar.find('input').props().className).toEqual(
      'makeStyles-inputStyle-3'
    )
  })

  it('renders a Box component with a prop name `component`', () => {
    expect(
      navBar
        .find(Box)
        .at(5)
        .props().component
    ).toEqual('img')
  })

  it('renders a Box component with a prop name `src`', () => {
    expect(
      navBar
        .find(Box)
        .at(5)
        .props().src
    ).toEqual('/vectors/search.svg')
  })

  it('renders a Box component with a prop name `alt`', () => {
    expect(
      navBar
        .find(Box)
        .at(5)
        .props().alt
    ).toEqual('search')
  })

  it('renders a NotificationBadge component', () => {
    expect(navBar.find(NotificationBadge).exists()).toBe(true)
  })

  it('renders a NotificationBadge component to match snapshots', () => {
    expect(navBar.find(NotificationBadge)).toMatchSnapshot()
  })

  it('renders a MessageBadge component', () => {
    expect(navBar.find(MessageBadge).exists()).toBe(true)
  })

  it('renders a MessageBadge component to match snapshot', () => {
    expect(navBar.find(MessageBadge)).toMatchSnapshot()
  })

  it('renders a MessageBadge component with a prop name `className`', () => {
    expect(navBar.find(MessageBadge).props().className).toEqual(
      'makeStyles-message-7'
    )
  })

  it('renders a NotificationBadge component with a prop name `className`', () => {
    expect(navBar.find(MessageBadge).props().img).toEqual(
      '/vectors/message.svg'
    )
  })

  it('renders a IconButton component', () => {
    expect(navBar.find(IconButton).exists()).toBe(true)
  })

  it('renders a IconButton component to match snapshot', () => {
    expect(navBar.find(IconButton)).toMatchSnapshot()
  })

  it('renders an IconButton with a prop name `className`', () => {
    expect(navBar.find(IconButton).props().className).toEqual(
      'makeStyles-iconButtonStyles-4'
    )
  })

  it('renders a Box with a prop name `component`', () => {
    expect(
      navBar
        .find(Box)
        .at(7)
        .props().component
    ).toEqual('img')
  })

  it('renders a Box component with a prop name `src` at node 7', () => {
    expect(
      navBar
        .find(Box)
        .at(7)
        .props().src
    ).toEqual('/vectors/avatar.svg')
  })

  it('renders a Box component with a prop name `alt` at node 7', () => {
    expect(
      navBar
        .find(Box)
        .at(7)
        .props().alt
    ).toEqual('avatar')
  })
})
