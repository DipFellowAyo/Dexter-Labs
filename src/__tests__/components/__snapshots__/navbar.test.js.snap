// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders a IconButton component to match snapshot 1`] = `
<WithStyles(ForwardRef(IconButton))
  className="makeStyles-iconButtonStyles-4"
>
  <Styled(MuiBox)
    alt="avatar"
    component="img"
    src="/vectors/avatar.svg"
  />
</WithStyles(ForwardRef(IconButton))>
`;

exports[`Navbar component renders a MessageBadge component to match snapshot 1`] = `
<MessageBadge
  className="makeStyles-message-7"
  img="/vectors/message.svg"
/>
`;

exports[`Navbar component renders a NotificationBadge component to match snapshots 1`] = `<NotificationBadge />`;

exports[`Navbar component renders correctly 1`] = `
<Styled(MuiBox)
  className="makeStyles-appBarStyles-1"
  component="nav"
>
  <Link
    to="/"
  >
    <Styled(MuiBox)
      alt="dexter"
      className="makeStyles-logo-5"
      component="img"
      src="/vectors/dexter.svg"
    />
    <Styled(MuiBox)
      alt="logo"
      className="makeStyles-mobile-6"
      component="img"
      src="/vectors/logo.svg"
    />
  </Link>
  <Styled(MuiBox)
    alignItems="center"
    display="flex"
  >
    <form>
      <Styled(MuiBox)
        className="makeStyles-searchField-2"
      >
        <input
          className="makeStyles-inputStyle-3"
          placeholder="Search"
          type="text"
        />
        <Styled(MuiBox)
          alt="search"
          component="img"
          src="/vectors/search.svg"
        />
      </Styled(MuiBox)>
    </form>
    <Styled(MuiBox)>
      <NotificationBadge />
      <MessageBadge
        className="makeStyles-message-7"
        img="/vectors/message.svg"
      />
      <WithStyles(ForwardRef(IconButton))
        className="makeStyles-iconButtonStyles-4"
      >
        <Styled(MuiBox)
          alt="avatar"
          component="img"
          src="/vectors/avatar.svg"
        />
      </WithStyles(ForwardRef(IconButton))>
    </Styled(MuiBox)>
  </Styled(MuiBox)>
</Styled(MuiBox)>
`;
